File: 2019.03.15/python/main.py
def productOfAllButOne(numList):
    result = []
    for i in range(len(numList)):
        result.append(1)
        for j in range(len(numList)):
            if i != j:
                result[i] *= numList[j]
    return result

if __name__ == "__main__":
    numList1 = [1, 2, 3, 4, 5]
    numList2 = [3, 2, 1]

    print(numList1, "->", productOfAllButOne(numList1))
    print(numList2, "->", productOfAllButOne(numList2))

File: 2019.03.15/javascript/main.js

function productOfAllButOne(numList) {
    let result = []
    for (let i=0; i < numList.length; i++) {
        result[i] = 1;
        for (let j=0; j < numList.length; j++) {
            if (i != j) result[i] *= numList[j];
        }
    }
    return result;
}

const numList1 = [1, 2, 3, 4, 5];
const numList2 = [3, 2, 1];

console.log("Input:", numList1);
console.log("Output:", productOfAllButOne(numList1));
console.log("Input:", numList2);
console.log("Output:", productOfAllButOne(numList2));